source/mainnet/smart-contracts/tutorials/counter/counter-contract.rst

```diff
@@ -25,215 +25,211 @@ Once you have set up the tools needed you are ready to create your smart contrac
 
     $cargo concordium init
 
-Select the ``default`` option from the menu.
+When prompted for which template to expand, select the ``default`` option. You will then be asked for a name for your project. In this example we'll use "counter", but you can choose whatever name you want.
 
-.. image:: images/select-default.png
-    :width: 100%
+The result is a basic skeleton of a smart contract. Initially, it has a ``State`` struct, an ``init`` function for creating new instances, an ``Error`` enum for custom errors, a ``view`` function to read the state, and a dummy ``receive`` function.
 
-Then it will ask for a name and a description of your project. Fill them in.
-The result is a basic skeleton of a smart contract.
-Initially, it has a ``State`` struct, an ``init`` function for creating new instances, an ``Error`` enum for custom errors, a ``view`` function, and a ``receive`` function.
-
-.. image:: images/contract.png
-    :width: 100%
-
-Add the counter to the state and i8 for integer. Then add the values ``OwnerError``, ``IncrementError``, and ``DecrementError`` to the ``Error`` enum, and specify the counter initial value as zero in the ``init`` function so the counter value starts from 0 when you create a new, fresh instance the contract. Your contract now looks like the example below.
+To build the counter smart contract, add a counter to the state as an 8-bit signed integer, the ``i8`` type. Then add the variants ``OwnerError`` and ``OverflowError`` to the ``Error`` enum, and specify the counter initial value as zero in the ``init`` function, so the counter value starts from zero when you create a new, fresh instance of the contract. The first part of your contract now looks like this:
 
 .. code-block:: rust
 
-    /// Your smart contract state.
-    #[derive(Serialize, SchemaType, Clone)]
+    /// The state of the smart contract.
+    #[derive(Serialize, SchemaType)]
     pub struct State {
-        // Your state
         counter: i8,
     }
 
-    /// Your smart contract errors.
-    #[derive(Debug, PartialEq, Eq, Reject, Serial, SchemaType)]
-    enum Error {
+    /// Errors that may be emitted by this smart contract.
+    #[derive(Debug, PartialEq, Eq, Reject, Serialize, SchemaType)]
+    pub enum Error {
         /// Failed parsing the parameter.
         #[from(ParseError)]
-        ParseParamsError,
-        /// Your error
+        ParseParams,
```

Comment:
Hmmm actually I just made this align to the default template, since that one uses `ParseParams`, so this is more like a template change than a change here in the tutorial.

Excluding the `Error` suffix is actually correct style. For instance, see this [Clippy lint](https://rust-lang.github.io/rust-clippy/master/index.html#/enum_variant_names). Generally enums should not repeat their names in the variants. Then you avoid these repetitions like `Error::SomeError`, and instead just have `Error::Some`. But of course the error should not necessarily be displayed like that when printed (i.e. its `Display` implementation probably ought to show `Error::SomeError` as `"Some error occurred" or something like that). It sounds like the real problem is that these other tools use the `Debug` representation of the error when really they shouldn't?

Reply:
You describe the recommended way of naming the Error enums in the Rust world. The point I am trying to make is that there is a custom error reporting mechanism in the Concordium smart contracts (using error schemas). This custom error reporting tries to facilitate reporting of errors when there is a chain of contract calls that reject/revert. Calling the variant "ParseParamsError" seems to make sense in the given Concordium smart contract environment in my opinion but we haven't been consistent with it as such I don't feel strongly about it and we can leave the current name as it is for now.